session_3/blogspot/app/controllers/articles_controller.rb

```diff
@@ -0,0 +1,78 @@
+class ArticlesController < ApplicationController
+  load_and_authorize_resource
+  
+  before_action :set_article, only: [:show, :edit, :update, :destroy]
+
+  # GET /articles
+  # GET /articles.json
+  def index
+    @articles = Article.all
+  end
+
+  # GET /articles/1
+  # GET /articles/1.json
+  def show
+    if !@article.public && !current_user.admin && @article.user_id != session[:user_id] && session[:private_articles_remaining]>0
+      session[:private_articles_remaining] -= 1
+    elsif !@article.public && session[:private_articles_remaining] < 1
```

Comment:
You can say `session[:private_articles_remaining] == 0` instead, that would be straightforward as no more articles are remaining.

Less than 1 makes me think about whether number of articles remaining could be fractional or negative.

Reply:
Ohk.. Just used '<' to avoid negative values incase